### DISCUSSION TITLE
See if `yield positions_to_evaluate` can be useful when writing ask-and-tell

### DISCUSSION THREAD

**Discussion by MichaelClerx** [pints-team/pints#1718] [Announcements]
See https://pythontips.com/2013/09/29/the-python-yield-keyword-explained/

Might be useful for some algorithms that are hard to break up, *or* might just make it harder to read :D

**Comment by MichaelClerx** [comment]
See also @martinjrobins 's cool work on NUTS

**Comment by martinjrobins** [comment]
I found this actually made the code easier to read, because you don't need to break up the algorithm across an ask/tell divide. Main problem with this is that Python's support for this is changing rapidly, the yield syntax for co-routines is deprecated and will be removed from 3.10 onward. And it is not supported by python v2

**Comment by MichaelClerx** [comment]
Possibly worth revisiting this at some point, but have turned it into a discussion rather than a pressing issue for now!
